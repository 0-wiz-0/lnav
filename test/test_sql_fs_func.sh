#! /bin/bash

run_test ./drive_sql "select basename('')"

check_output "basename('') is not '.'" <<EOF
Row 0:
  Column basename(''): .
EOF

run_test ./drive_sql "select basename('/')"

check_output "basename('/') is not '/'" <<EOF
Row 0:
  Column basename('/'): /
EOF

run_test ./drive_sql "select basename('//')"

check_output "basename('//') is not '/'" <<EOF
Row 0:
  Column basename('//'): /
EOF

run_test ./drive_sql "select basename('/foo')"

check_output "basename('/foo') is not 'foo'" <<EOF
Row 0:
  Column basename('/foo'): foo
EOF

run_test ./drive_sql "select basename('/foo/')"

check_output "basename('/foo/') is not 'foo'" <<EOF
Row 0:
  Column basename('/foo/'): foo
EOF

run_test ./drive_sql "select basename('/foo///')"

check_output "basename('/foo///') is not 'foo'" <<EOF
Row 0:
  Column basename('/foo///'): foo
EOF

run_test ./drive_sql "select basename('foo')"

check_output "basename('foo') is not 'foo'" <<EOF
Row 0:
  Column basename('foo'): foo
EOF


run_test ./drive_sql "select dirname('')"

check_output "dirname('') is not '.'" <<EOF
Row 0:
  Column dirname(''): .
EOF

run_test ./drive_sql "select dirname('foo')"

check_output "dirname('foo') is not '.'" <<EOF
Row 0:
  Column dirname('foo'): .
EOF

run_test ./drive_sql "select dirname('foo///')"

check_output "dirname('foo///') is not '.'" <<EOF
Row 0:
  Column dirname('foo///'): .
EOF

run_test ./drive_sql "select dirname('/foo/bar')"

check_output "dirname('/foo/bar') is not '/foo'" <<EOF
Row 0:
  Column dirname('/foo/bar'): /foo
EOF

run_test ./drive_sql "select dirname('/')"

check_output "dirname('/') is not '/'" <<EOF
Row 0:
  Column dirname('/'): /
EOF

run_test ./drive_sql "select dirname('/foo')"

check_output "dirname('/foo') is not '/'" <<EOF
Row 0:
  Column dirname('/foo'): /
EOF

run_test ./drive_sql "select dirname('/foo//')"

check_output "dirname('/foo//') is not '/'" <<EOF
Row 0:
  Column dirname('/foo//'): /
EOF

run_test ./drive_sql "select dirname('foo//')"

check_output "dirname('foo//') is not '.'" <<EOF
Row 0:
  Column dirname('foo//'): .
EOF


run_test ./drive_sql "select joinpath()"

check_output "joinpath() is not ''" <<EOF
Row 0:
  Column joinpath(): (null)
EOF

run_test ./drive_sql "select joinpath('foo')"

check_output "joinpath('foo') is not ''" <<EOF
Row 0:
  Column joinpath('foo'): foo
EOF

run_test ./drive_sql "select joinpath('foo', 'bar', 'baz')"

check_output "joinpath('foo', 'bar', 'baz') is not ''" <<EOF
Row 0:
  Column joinpath('foo', 'bar', 'baz'): foo/bar/baz
EOF

run_test ./drive_sql "select joinpath('foo', 'bar', 'baz', '/root')"

check_output "joinpath('foo', 'bar', 'baz', '/root') is not ''" <<EOF
Row 0:
  Column joinpath('foo', 'bar', 'baz', '/root'): /root
EOF
