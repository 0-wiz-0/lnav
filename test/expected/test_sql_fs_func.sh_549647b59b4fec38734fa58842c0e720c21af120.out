Row 0:
  Column realpath('sql_fs_realpath_test.lnk'): {builddir}/drive_sql
